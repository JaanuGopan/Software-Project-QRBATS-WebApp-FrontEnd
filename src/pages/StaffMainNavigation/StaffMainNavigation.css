@media screen and (max-width: 1800px) {
    .Webname {
      font-size: x-large;
    }
    .Staff-Logo{
      height: 4.5rem;
    }
    .ProfileButton{
      height: 2.5rem;
    }
    .ProfileName{
      font-weight: bold;
    }
    .Staff-Sidebar{
      width: 250px;
    }
    .buttonIcon{
      height: 1.5rem;
      width: 1.5rem;
    }
    .SideBarButton{
      height: 3rem;
      padding-left: 2vw;
    }
    .SideBar{
      margin-top: 2vh;
      width: 100%;
    }
    .SideBarLabel{
      font-size: larger;
      padding: 1vw;
      margin-bottom: 0px;
    }
}

@media screen and (max-width: 768px) {
    .Webname {
        font-size: large;
    }
    .Staff-Logo{
      height: 3.5rem;
    }
    .ProfileButton{
      height: 2.5rem;
    }
    .ProfileName{
      font-weight: bold;
    }
    .Staff-Sidebar{
      width: 60px;
    }
    .Sidebarlogo{
      visibility: hidden;
    }
    .buttonIcon{
      height: 2rem;
      width: 2rem;
    }
    .SideBarLabel{
      visibility: hidden;
      font-size: 0px;
      padding: 0px;
      margin-bottom: 0px;
    }
    .SideBar{
      margin-top: 5vh;
      width: 100%;
      position: absolute;
      display: flex;
      flex-direction: column;
      align-items: center;
    }
}

@media screen and (max-width: 420px) {
    .Webname {
        font-size: small;
    }
    .ProfileName{
        font-size: small;
    }
    .Staff-Logo{
      height: 3rem;
    }
    .ProfileButton{
      height: 2rem;
    }
    .ProfileName{
      font-weight: normal;
    }
}

.staff-Main{
  display: flex;
  flex-direction: column;
  width: 100vw;
  align-items: stretch;
  justify-content: flex-start;
  height: 100vh;
}

.Staff-Navbar{
  background-color: #0063A0;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 5rem;
  width: 100%;
  position: relative;
  flex-wrap: nowrap;
}

.Staff-Icon{
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: nowrap;
    height: 4.5rem;
    width: 35vw;
    padding-left: 2vw;
}

.Staff-Logo {
    padding: 5px;
    margin-right: 1vw;
  }

.Webname {
    color: white;
    letter-Spacing: 0.5rem;
    margin-Left: 0.5vw;
    margin-Bottom: 0px;
    font-weight: bold;
}

.ProfileButton{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
    flex-wrap: nowrap;
    border: 1px solid white;
    margin-right: 2vw;
    cursor: pointer;
    border-radius: 30px;
}

.ProfileButton:hover{
    background-color: white;
    border-color: #0063A0;
    .ProfileName{
      color: #0063A0;
    }
}

.ProfileName{
    color: white;
    margin-right: 0.5vw;
    margin-Bottom: 0px;
    padding: 10px;
}

.circle {/* Adjust size as needed */
    height: 100%; /* Adjust size as needed */
    border-radius: 30px; /* Makes it a circle */
    overflow: hidden; /* Clips the content to the shape of the container */
    border: 2px solid #ccc; /* Optional: Adds a border */
}

.ProfilePicture {
    height: 100%;
}

.staff-Submain{
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: flex-start;
    height: 100vh;
    position: relative;
}

.Staff-Sidebar{
  background: linear-gradient(#0063A0, rgba(0,100,160,0.8));
  display: flex;
  position: relative;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  border-top: 5px solid white;
  border-top-right-radius: 20px;
  z-index: 1;
  height:-webkit-fill-available;
}

.Sidebarlogo{
  height: 130px;
  width: 150px;
  padding: 5px;
  margin-top: 30px;
}

.SideBarButton{
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  color: white;
  width: 100%;
  cursor: pointer;
}

.SideBarButton:hover{
  background-color: #025587;
  border-radius: 10px;
  border: 1px solid white;
}

.SideBarButton.active {
  background-color: #f0f0f0;
  color: #333;
  border-left: 3px solid #007bff; /* Example of adding a border to indicate active state */
}

.menuButton{
  position: absolute;
  color:'black';
  margin-top: 5rem;
  z-index: 1;
  cursor: pointer;
  background-color:#ccc;
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  padding: 5px;
  box-shadow: 2px 2px 2px 0px rgba(27, 37, 49, 0.5);
}

.menuButton:hover{
  background-color: darkgray;
}

.closeButton{
  color: white;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: flex-end;
  margin-left: 75%;
  margin-top: 10px;
  padding: 5px;
  border-radius: 50px;
}
.closeButton:hover{
  background-color: #025587;
}

.Staff-Subpage{
  display: block;
  position: relative;
  background-color: aqua;
  height:-webkit-fill-available;
  width:-webkit-fill-available;
}