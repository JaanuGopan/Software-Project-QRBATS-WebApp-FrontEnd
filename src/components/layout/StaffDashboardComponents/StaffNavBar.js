import React from "react";
import logo from "../../../assets/Images/logo/logo_white.png";
import "../../../pages/StaffMainNavigation/StaffMainNavigation.css";
import ProfileButton from "./ProfileButton";

const StaffNavBar = ({ setIndex }) => {
  return (
    <div className="Staff-Navbar">
      <div className="Staff-Icon">
        <img className="Staff-Logo" src={logo} alt="" />
        <p className="Webname">SKY TICKER</p>
      </div>
      <ProfileButton setIndex={setIndex} />
    </div>
  );
};

export default StaffNavBar;
