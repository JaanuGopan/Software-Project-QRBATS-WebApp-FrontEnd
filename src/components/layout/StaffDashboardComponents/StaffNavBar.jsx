import React from 'react';
import logo from '../../../assets/Images/logo/logo_white.png';
import '../../../pages/StaffMainNavigation/StaffMainNavigation.css';
import ProfileButton from './ProfileButton';
import { useNavigate } from 'react-router-dom';

const StaffNavBar = ({ setIndex }) => {
  const navigator = useNavigate();
  return (
    <div className="Staff-Navbar">
      <div className="Staff-Icon">
        <img
          className="Staff-Logo"
          style={{ cursor: 'pointer' }}
          src={logo}
          alt="Sky Ticker"
          onClick={() => navigator('/')}
        />
        <p className="Webname">SKY TICKER</p>
      </div>
      <ProfileButton setIndex={setIndex} />
    </div>
  );
};

export default StaffNavBar;
