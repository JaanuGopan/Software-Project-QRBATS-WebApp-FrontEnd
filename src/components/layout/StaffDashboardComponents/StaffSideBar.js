import React from "react";
import { IoMdCloseCircleOutline } from "react-icons/io";
import { IoSettings } from "react-icons/io5";
import { FaUserTie } from "react-icons/fa";
import { PiStudentBold } from "react-icons/pi";
import { MdOutlineEventNote } from "react-icons/md";
import { IoLogOut } from "react-icons/io5";
import "../../../pages/StaffMainNavigation/StaffMainNavigation.css";
import logo from "../../../assets/Images/logo/logo_white.png";
import Sidebarbutton from "./Sidebarbutton";
import { AiFillDashboard } from "react-icons/ai";

const StaffSideBar = ({ handleclose }) => {
  return (
    <div class="Staff-Sidebar">
      <div className="closeButton" onClick={handleclose}>
        <IoMdCloseCircleOutline />
      </div>
      <img src={logo} className="Sidebarlogo" />
      <div className="SideBar">
        <Sidebarbutton
          title={"Dashboard"}
          titlewithiconicon={<AiFillDashboard className="buttonIcon" />}
        />
        <Sidebarbutton
          title={"Event"}
          titlewithiconicon={<MdOutlineEventNote className="buttonIcon" />}
        />
        <Sidebarbutton
          title={"Staff"}
          titlewithiconicon={<FaUserTie className="buttonIcon" />}
        />
        <Sidebarbutton
          title={"Student"}
          titlewithiconicon={<PiStudentBold className="buttonIcon" />}
        />
        <Sidebarbutton
          title={"Setting"}
          titlewithiconicon={<IoSettings className="buttonIcon" />}
        />
        <Sidebarbutton
          title={"Logout"}
          titlewithiconicon={<IoLogOut className="buttonIcon" />}
        />
      </div>
    </div>
  );
};

export default StaffSideBar;
