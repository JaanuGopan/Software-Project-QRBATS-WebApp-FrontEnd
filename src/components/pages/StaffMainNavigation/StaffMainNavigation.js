import React from 'react';
import StaffNavBar from '../../layout/StaffDashboardComponents/StaffSideBar';
import StaffSideBar from '../../layout/StaffDashboardComponents/StaffSideBar';

const StaffMainNavigation = () => {
  return (
    <div className='staff-Main'>
        <StaffNavBar/>
      <div className='staff-Submain'>
        <StaffSideBar/>
        <div className='staff-SubPage'>
            SubPages
        </div>
      </div>
    </div>
  );
}

export default StaffMainNavigation;