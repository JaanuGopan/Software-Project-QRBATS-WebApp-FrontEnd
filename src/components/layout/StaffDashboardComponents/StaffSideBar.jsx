import React from 'react';
import { AiFillDashboard } from 'react-icons/ai';
import { FaUserTie } from 'react-icons/fa';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import { IoLogOut, IoSettings } from 'react-icons/io5';
import { MdOutlineEventNote } from 'react-icons/md';
import { PiStudentBold } from 'react-icons/pi';
import logo from '../../../assets/Images/logo/logo_white.png';
import '../../../pages/StaffMainNavigation/StaffMainNavigation.css';
import Sidebarbutton from './Sidebarbutton';

const StaffSideBar = ({ handleclose }) => {
  return (
    <div class="Staff-Sidebar">
      <div className="closeButton" onClick={handleclose}>
        <IoMdCloseCircleOutline />
      </div>
      <img src={logo} className="Sidebarlogo" />
      <div className="SideBar">
        <Sidebarbutton title={'Dashboard'} icon={<AiFillDashboard className="buttonIcon" />} />
        <Sidebarbutton title={'Event'} icon={<MdOutlineEventNote className="buttonIcon" />} />
        <Sidebarbutton title={'Staff'} icon={<FaUserTie className="buttonIcon" />} />
        <Sidebarbutton title={'Student'} icon={<PiStudentBold className="buttonIcon" />} />
        <Sidebarbutton title={'Settings'} icon={<IoSettings className="buttonIcon" />} />
        <Sidebarbutton title={'Logout'} icon={<IoLogOut className="buttonIcon" />} />
      </div>
    </div>
  );
};

export default StaffSideBar;
