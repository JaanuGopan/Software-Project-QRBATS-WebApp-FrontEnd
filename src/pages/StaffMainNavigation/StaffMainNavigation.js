import React, { useState } from "react";
import "./StaffMainNavigation.css";
import { PiListDashesFill } from "react-icons/pi";
import StaffNavBar from "../../components/layout/StaffDashboardComponents/StaffNavBar";
import StaffSideBar from "../../components/layout/StaffDashboardComponents/StaffSideBar";

const StaffMainNavigation = () => {
  const [isHidden, setIsHidden] = useState(false);
  const handleshow = () => {
    setIsHidden(false);
  };
  const handleclose = () => {
    setIsHidden(true);
  };
  return (
    <div className="staff-Main">
      <div className="menuButton" onClick={handleshow}>
        <PiListDashesFill size={"30px"} />
      </div>
      <StaffNavBar />
      <div class="staff-Submain">
        {!isHidden && <StaffSideBar handleclose={handleclose} />}
        <div className="Staff-Subpage">
          <div>SubPages</div>
        </div>
      </div>
    </div>
  );
};

export default StaffMainNavigation;
