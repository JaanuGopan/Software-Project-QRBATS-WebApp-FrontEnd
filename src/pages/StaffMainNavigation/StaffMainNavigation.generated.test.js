import renderer from 'react-test-renderer';
import React, { useState } from "react";

import { PiListDashesFill } from "react-icons/pi";
import StaffNavBar from "../../components/layout/StaffDashboardComponents/StaffNavBar";
import StaffSideBar from "../../components/layout/StaffDashboardComponents/StaffSideBar";
import StaffMainNavigation from "./StaffMainNavigation";

jest.mock("./StaffMainNavigation.css");
jest.mock("react-icons/pi");
jest.mock("../../components/layout/StaffDashboardComponents/StaffNavBar");
jest.mock("../../components/layout/StaffDashboardComponents/StaffSideBar");

const renderTree = tree => renderer.create(tree);
describe('<StaffMainNavigation>', () => {
  it('should render component', () => {
    expect(renderTree(<StaffMainNavigation 
    />).toJSON()).toMatchSnapshot();
  });
  
});