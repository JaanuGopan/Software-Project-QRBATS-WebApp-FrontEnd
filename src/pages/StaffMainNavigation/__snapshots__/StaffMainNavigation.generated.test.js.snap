// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffMainNavigation> should render component 1`] = `
<div
  className="staff-Main"
>
  <div
    className="menuButton"
    onClick={[Function]}
  />
  <div
    class="staff-Submain"
  >
    <div
      className="Staff-Subpage"
    >
      <div>
        SubPages
      </div>
    </div>
  </div>
</div>
`;
